### PR TITLE
[stable] Add dmd.argtypes_aarch64 for the AAPCS64 ABI

### PR DESCRIPTION
**Pull request by kinke**
From https://github.com/ldc-developers/ldc/pull/3393; only useful for LDC for now.

### CODE REVIEW

File: src/dmd/mtype.h [hunk 1]
```diff
@@ -815,6 +815,8 @@ class TypeClass : public Type
 class TypeTuple : public Type
 {
 public:
+    static TypeTuple *empty;    // don't modify!
```
Comment (ibuclaw) [inline src/dmd/mtype.h:818]: Needs a better comment here.

File: src/dmd/aggregate.h [hunk 2]
```diff
@@ -196,6 +195,9 @@ class StructDeclaration : public AggregateDeclaration
 
     StructDeclaration *isStructDeclaration() { return this; }
     void accept(Visitor *v) { v->visit(this); }
+
+    d_size_t numArgTypes() const;
+    Type *argType(d_size_t index);
```
Comment (ibuclaw) [inline src/dmd/aggregate.h:200]: Please use `uint`.

File: src/dmd/dstruct.d [hunk 3]
```diff
@@ -603,6 +594,16 @@ extern (C++) class StructDeclaration : AggregateDeclaration
     {
         v.visit(this);
     }
+
+    final size_t numArgTypes() const
```
Comment (ibuclaw) [inline src/dmd/dstruct.d:598]: Please use `uint`.

File: src/dmd/dstruct.d [hunk 4]
```diff
@@ -603,6 +594,16 @@ extern (C++) class StructDeclaration : AggregateDeclaration
...
+    {
+        return argTypes && argTypes.arguments ? argTypes.arguments.dim : 0;
+    }
+
+    final Type argType(size_t index)
```
Comment (ibuclaw) [inline src/dmd/dstruct.d:603]: Please use `uint`.

**Review by ibuclaw — CHANGES_REQUESTED** [review]
A minor ABI nit, but otherwise OK.

File: src/dmd/mtype.h (hunk 1 above)
Comment (kinke) [reply src/dmd/mtype.h]: FWIW, changed to `'logically immutable' cached global - don't modify (neither pointer nor pointee)!`.

Comment (ibuclaw) [reply src/dmd/mtype.h]: I was thinking more along the lines of *what* it stores, rather than the why, but ok.

Comment (kinke) [reply src/dmd/mtype.h]: I thought its name would make that clear.

**Review by ibuclaw — APPROVED** [review]

**Pull request merged**